workspaces/mcp-integrations/plugins/mcp-actions-backend-backend/src/services/McpService.ts

```diff
@@ -0,0 +1,101 @@
+/*
+ * Copyright 2025 The Backstage Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+import { BackstageCredentials } from '@backstage/backend-plugin-api';
+import { Server as McpServer } from '@modelcontextprotocol/sdk/server/index.js';
+import {
+  ListToolsRequestSchema,
+  CallToolRequestSchema,
+} from '@modelcontextprotocol/sdk/types.js';
+import { JsonObject } from '@backstage/types';
+import { ActionsService } from '@backstage/backend-plugin-api/alpha';
+// This does not seem to work when the plugin is packaged as a dynamic plugin -- it says package.json does not exist
+// This likely warrants some more debugging, but for now just hardcode the version
```

Comment:
I wonder if the dynamic plugin process needs to have the package registered up on npm 

feels somewhat analogous when I tried to leverage our entity provide plugin as a dependency of the techdoc url reader service plugin